runhouse/resources/functions/aws_lambda.py

```diff
@@ -0,0 +1,887 @@
+        >>> res = summer(5, 8)  # returns "13". (It returns str type because of AWS API)
+
+        >>> # Load function from above
+        >>> reloaded_function = rh.lambda_function(name="my_func")
+
+        >>> # Pass in the function itself when creating the Lambda
+        >>> lambdas_func = rh.lambda_function(fn=summer, name="lambdas_func")
+
+    """
+    # TODO: [SB] in the next phase, maybe add the option to create func from git.
+    if name and not any(
+        [paths_to_code, handler_function_name, runtime, fn, args_names]
+    ):
+        # Try reloading existing function
+        return LambdaFunction.from_name(name=name)
+
+    if not (fn or (handler_function_name and paths_to_code)):
+        raise RuntimeError(
+            "Please provide a callable function OR path to handler function and its name "
+            + "in order to create a Lambda function."
+        )
+    # Env setup.
+    if env is not None and not isinstance(env, Env):
+        original_env = copy.deepcopy(env)
+        if isinstance(original_env, dict) and "env_vars" in original_env.keys():
+            env = _get_env_from(env["reqs"]) or Env(
+                working_dir="../functions/", name=Env.DEFAULT_NAME
+            )
+        elif isinstance(original_env, str):
+            env = _get_env_from(env)
+        else:
+            env = _get_env_from(env) or Env(
+                working_dir="../functions/", name=Env.DEFAULT_NAME
+            )
+
+    elif env is None:
+        env = Env(
+            reqs=[],
+            env_vars={"HOME": LambdaFunction.HOME_DIR},
+            name=Env.DEFAULT_NAME,
+        )
+
+    if isinstance(env, Env) and "HOME" not in env.env_vars.keys():
+        env.env_vars["HOME"] = LambdaFunction.HOME_DIR
+
+    # extract function pointers, path to code and arg names from callable function.
+    if isinstance(fn, Callable):
+        handler_function_name = fn.__name__
+        fn_pointers = Function._extract_pointers(
+            fn, reqs=[] if env is None else env.reqs
+        )
+        paths_to_code = _paths_to_code_from_fn_pointers(fn_pointers)
+        args_names = [param.name for param in inspect.signature(fn).parameters.values()]
+        if name is None:
+            name = fn.__name__
+
+    else:
+
+        # ------- arguments validation -------
+        if paths_to_code is None or len(paths_to_code) == 0:
+            raise RuntimeError("Please provide a path to the lambda handler file.")
+        if isinstance(env, str) and "requirements.txt" in env:
+            paths_to_code.append(Path(env).absolute())
+        if handler_function_name is None or len(handler_function_name) == 0:
+            raise RuntimeError(
+                "Please provide the name of the function that should be executed by the lambda."
+            )
+        if args_names is None:
+            # Parsing the handler file and extracting the arguments names of the handler function.
+            file_path = paths_to_code[0]
```

Comment:
If we can would be cleaner to use the inspect library to handle this extraction for us - something along the lines of: 

```
            spec = importlib.util.spec_from_file_location(module_name, file_path)
            module = importlib.util.module_from_spec(spec)
            spec.loader.exec_module(module)

            # Extract the function and its arguments
            func = getattr(module, func_name, None)
            args_names = inspect.getfullargspec(func).args
```